### PR TITLE
AIBrix Workload Generator / Scenario Simulator

### PR DESCRIPTION
**Pull request by happyandslow**
## Pull Request Description
[Please provide a clear and concise description of your changes here]

## Related Issues
Resolves: #[Insert issue number(s)]

**Important: Before submitting, please complete the description above and review the checklist below.**

---

<details>
<summary><strong>Contribution Guidelines (Expand for Details)</strong></summary>

<p>We appreciate your contribution to aibrix! To ensure a smooth review process and maintain high code quality, please adhere to the following guidelines:</p>

<h3>Pull Request Title Format</h3>
<p>Your PR title should start with one of these prefixes to indicate the nature of the change:</p>
<ul>
    <li><code>[Bug]</code>: Corrections to existing functionality</li>
    <li><code>[CI]</code>: Changes to build process or CI pipeline</li>
    <li><code>[Docs]</code>: Updates or additions to documentation</li>
    <li><code>[API]</code>: Modifications to aibrix's API or interface</li>
    <li><code>[CLI]</code>: Changes or additions to the Command Line Interface</li>
    <li><code>[Misc]</code>: For changes not covered above (use sparingly)</li>
</ul>
<p><em>Note: For changes spanning multiple categories, use multiple prefixes in order of importance.</em></p>

<h3>Submission Checklist</h3>
<ul>
    <li>[ ] PR title includes appropriate prefix(es)</li>
    <li>[ ] Changes are clearly explained in the PR description</li>
    <li>[ ] New and existing tests pass successfully</li>
    <li>[ ] Code adheres to project style and best practices</li>
    <li>[ ] Documentation updated to reflect changes (if applicable)</li>
    <li>[ ] Thorough testing completed, no regressions introduced</li>
</ul>

<p>By submitting this PR, you confirm that you've read these guidelines and your changes align with the project's contribution standards.</p>

</details>

### CODE REVIEW

File: benchmarks/autoscaling/benchmark.py [hunk 1]
```diff
@@ -235,69 +235,96 @@ def sample_sharegpt_requests(
...
+                        asyncio.run(benchmark_requests(openai_clients, openai_endpoints, each_input_requests, len(each_input_requests),
+                                                    len(each_input_requests), args.output_file_path))
+                        # wait until passing args.w_interval_sec
+                        wait_time = next_start - time.time()
+                        if wait_time > 0:
+                            time.sleep(wait_time)
+                        next_start += interval_sec
+                end_time = time.time()
+        except FileNotFoundError:
+            print("The file was not found")
```
Comment (Jeffwan) [inline benchmarks/autoscaling/benchmark.py]: minor: Let's use logging for consistency

File: benchmarks/autoscaling/benchmark.py [hunk 2]
```diff
@@ -235,69 +235,96 @@ def sample_sharegpt_requests(
...
-                time.sleep(wait_time)
-            next_start += interval_sec
-    end_time = time.time()
-    logging.info(f"Benchmark completed in {end_time - start_time:.2f} seconds")
+        try:
+            interval_sec = args.w_interval_sec
+            with open(args.workload, 'r') as file:
+                generated_workloads = json.load(file)
+                start_time = time.time()    
+                next_start = start_time + interval_sec
```
Comment (Jeffwan) [inline benchmarks/autoscaling/benchmark.py]: interval_sec is fixed value?

File: benchmarks/autoscaling/benchmark.py [hunk 3]
```diff
@@ -235,69 +235,96 @@ def sample_sharegpt_requests(
...
+            workloads = generate_workload(
+                input_requests,
+                A=args.w_A, B=args.w_B, sigma=args.w_sigma, only_rise=args.w_only_rise,
+                period=args.w_period, duration_sec=args.w_duration_sec, interval_sec=interval_sec,
+            )
+            # if you want to see the workload traffic trend
+            plot_workload({"llm": workloads}, interval_sec=interval_sec,
+                        output_path=f"workload_plot/{identifier}.png")
+            next_start = start_time + interval_sec
+            for idx, each_input_requests in enumerate(workloads):
```
Comment (Jeffwan) [inline benchmarks/autoscaling/benchmark.py]: the client side code is exact same as generator mode. Should we refactor this part?

either accept generated trace or leverage  `generate_workload` to generate the load, then starts the benchmarking.

File: benchmarks/generator/generate.sh [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
```
Comment (Jeffwan) [inline benchmarks/generator/generate.sh]: I suggest to change this file to a README.md

File: benchmarks/lora/benchmark.py [hunk 5]
```diff
@@ -219,36 +220,61 @@ def main(args):
...
 
-    if args.dataset_path is not None:
-        logging.info(f"Start to sample {num_prompts} prompts from {args.dataset_path}")
-        input_requests = sample_sharegpt_requests(
-            dataset_path=args.dataset_path,
-            num_requests=num_prompts,
-            tokenizer=tokenizer,
-            fixed_output_len=args.sharegpt_output_len,
-        )
+    if args.workload is not None:
```
Comment (Jeffwan) [inline benchmarks/lora/benchmark.py]: lot of clients/experiments has this similar methods. can we refactor this part? If you do not like to make it now, let's have a brief chat.

Comment (happyandslow) [reply benchmarks/lora/benchmark.py]: This change has been removed

File: benchmarks/generator/generate.sh (hunk 4 above)
Comment (happyandslow) [reply benchmarks/generator/generate.sh]: Fixed

File: benchmarks/autoscaling/benchmark.py (hunk 3 above)
Comment (happyandslow) [reply benchmarks/autoscaling/benchmark.py]: This change has been removed

File: benchmarks/autoscaling/benchmark.py (hunk 2 above)
Comment (happyandslow) [reply benchmarks/autoscaling/benchmark.py]: This change has been removed

File: benchmarks/autoscaling/benchmark.py (hunk 1 above)
Comment (happyandslow) [reply benchmarks/autoscaling/benchmark.py]: This change has been removed

**Pull request merged**